docs/signing-spec.md

```diff
@@ -22,22 +30,59 @@ Asymmetric cryptography is nice for this because a signer can distribute a publi
 ## Signing and Roles
 
 Signers of an invoice MUST have a role. The following roles are defined by this specification.
-Agents SHOULD initiate a warning and MAY initiate a failure if another role is encountered.
+Agents SHOULD initiate a warning and MAY initiate a failure if a user's expected role does not match the given role.
```

Comment:
Should this be stronger and suggest that clients `SHOULD` initiate a failure if the roles don't match?

Reply:
I am not sure. The issue for me is that the user maintains their own keychain. If they choose to put a key in there as a `approver` key, but that person creates a packages, I do think that the tooling should let the user know... but I don't know whether it should fail.

Now having said that, I think I changed my mind. It does seem like the right thing to do would be to say, "This package was signed with a `creator` key that you only allow to use to `approve`. Stopping. To fix this, edit your keychain."

(I mean, obviously in a more user-friendly way... but the intent is the same)

Reply:
Changed